design_docs/oscode/oscode_sundials.md

```diff
@@ -3,50 +3,59 @@
 # Summary
 [summary]: #summary
 
-This design document describes the motivation and architecture for adding the OSCODE numerical routine for the efficient solution of one dimensional, second order, ordinary differential equations with rapidly oscillating solutions. The proposed solver will will operate with differential equations of the form 
+This design document describes the motivation and architecture for adding the OSCODE numerical routine for the efficient solution of one dimensional, second order, linear, homogeneous ordinary differential equations with rapidly oscillating solutions. The proposed solver will will operate with differential equations of the form 
 
 $$ \ddot{x} + 2 \gamma(t) \dot{x}(t) + \omega^2(t)x(t) = 0 $$
 
-where $\gamma$ and $\omega$ may or may not be expressed as a closed form function of time.
+where $\gamma$ and $\omega$ may or may not be expressed as a closed form function of time. We further require $\gamma$ and $\omega$ to be real-valued. If $\omega$ is large, the solution may be highly oscillatory, and standard (polynomial-based) numerical methods will require $\mathcal{O}(\omega) timesteps/discretization points. In regimes where $\omega$ is large and smooth, OSCODE exploits an asympotic approximation to reduce this computational cost to $\mathcal{O}(1)$ ($\omega$-independent). In other regimes of the solution interval, OSCODE behaves as a Runge--Kutta solver, and is thus robust to changes in the behavior in the solution from oscillatory to non-oscillatory.
+
+<!--- We should cite the oscode papers for details, and the JOSS paper! --->
```

Comment:
Yes I included a link to the original paper at the bottom but should also include links to the JOSS paper / code / docs